dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/transport/http/URLConnectionHTTPConduit.java

```diff
@@ -130,6 +132,20 @@ private HttpURLConnection createConnection(Message message, Address address, HTT
     @Override
     protected void setupConnection(Message message, Address address, HTTPClientPolicy csPolicy) throws IOException {
         HttpURLConnection connection = createConnection(message, address, csPolicy);
+        
+        // If the Expect header is set to "100-continue" then currently the JDK
+        // will not handle the returned 100 and will pass that message on the JAXRS (that is expecting
+        // the 200, not the 100).   This causes issues and potentially hangs.   To work around this
+        // issue we will allow URLConnectionHttpConduit.setFixedLengthStreamingMode() to be used.  
+        // Normally FixedLengthStreamMode is disabled due to a JDK performance issue [CXF-6227].
+        Map<String, List<String>> headers = CastUtils.cast((Map<?, ?>)message.get(Message.PROTOCOL_HEADERS));
+        List<String> expectList = headers.get("Expect");
+        
+        if ((expectList != null) && (expectList.contains("100-continue"))) {
```

Comment:
According to the HTTP spec, the value of the `Expect` header is case-insensitive.  I think you'll need to iterate over the header values and check for for `value.equalsIgnoreCase("100-continue")`.

Reply:
Also, this seems absurd to me...  we're already handicapping CXF because of a JDK performance issue - so now we're making performance worse, on _every request_ to workaround another JDK bug.  The JDK needs to fix this, not us.

Reply:
@andymc12  I agree,  however switching the JDK may not be a viable solution since it would require us to add potentially performance impacting code to detect the JDK and do something different depending.